source/CMakeLists.txt

```diff
@@ -92,10 +94,18 @@ else()
 endif()
 
 if(USE_OPENCV)
-    find_package(OpenCV 4.1.0 COMPONENTS core highgui calib3d aruco)
-    if(NOT OpenCV_FOUND)
-        message(FATAL_ERROR "OpenCV not found. Please point OpenCV_DIR to the directory of your OpenCV installation (containing the file OpenCVConfig.cmake), or disable the OpenCV samples  with -DUSE_OPENCV=OFF.")
-    endif()
+	if(USE_ARUCO)
+		find_package(OpenCV 4.1.0 COMPONENTS core highgui calib3d aruco)
```

Comment:
I suggest avoiding duplicating the opencv modules here and below. Or it will be easy for someone to update one and forget the other when we add a dependency on a new module. Something like this maybe?

```diff
diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index 914ca18..a61efec 100644
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -94,14 +94,15 @@ else()
 endif()
 
 if(USE_OPENCV)
+    set(OPENCV_COMMON_MODULES core highgui calib3d)
        if(USE_ARUCO)
-               find_package(OpenCV 4.1.0 COMPONENTS core highgui calib3d aruco)
+               find_package(OpenCV 4.1.0 COMPONENTS ${OPENCV_COMMON_MODULES} aruco)
                if(NOT OpenCV_FOUND)
                        message(FATAL_ERROR "OpenCV not found. Please point OpenCV_DIR to the directory of your OpenCV installation (containing the file OpenCVConfig.cmake), or disable the ArUco samples  with -DUSE_ARUCO=OFF.")
                endif()
        else()
            disable_samples("ArUco")
-           find_package(OpenCV 4.1.0 COMPONENTS core highgui calib3d)
+           find_package(OpenCV 4.1.0 COMPONENTS ${OPENCV_COMMON_MODULES})
                if(NOT OpenCV_FOUND)
                        message(FATAL_ERROR "OpenCV not found. Please point OpenCV_DIR to the directory of your OpenCV installation (containing the file OpenCVConfig.cmake), or disable the OpenCV samples  with -DUSE_OPENCV=OFF.")
                endif()
```

Reply:
done